komodoenv/__main__.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import print_function
+import os
+import sys
+import argparse
+import logging
+from shutil import rmtree
+from pathlib import Path
+from komodoenv.creator import Creator
+from komodoenv.statfs import is_nfs
+from colors import red, yellow, strip_color
+
+
+def get_text(release_path):
+    name = release_path.name
+
+    if name.startswith("bleeding"):
+        return yellow(
+            "Warning: Tracking a bleeding release of komodo. It changes every day. "
+            "You will need to recreate komodoenv in order to use new "
+            "executables in komodo.\n"
+        )
+    elif name.startswith("unstable"):
+        return yellow("Warning: Tracking an unstable release of komodo.\n")
+    elif name.startswith("testing"):
+        return yellow("Warning: Tracking a testing release of komodo.\n")
+    elif name.startswith("stable"):
+        return "Tracking a stable release of komodo.\n"
+    else:
+        return yellow(
+            "Warning: Tracking an untracked release. It will not receive updates. "
+            "This'll require recreating komodoenv to get updated software.\n"
+        )
+
+
+def autodetect(root):
+    """Autodetect komodo release heuristically"""
+    name = os.environ.get("KOMODO_RELEASE", "")
+    if not name:
+        return None
+
+    name = os.environ["KOMODO_RELEASE"]
+
+    release_root = root / os.environ["KOMODO_RELEASE"]
+    if not release_root.is_dir():
+        return None
+
+    for mode in "stable", "testing", "unstable", "bleeding":
+        for suffix in "", "-py2", "-py27", "-py3", "-py36":
```

Comment:
Matrix builds will yield `2020.10.00-py27` which is just a switch. I think the heuristic here is to insist on a concrete build if you do not find a `root` folder. A switch will not contain a `root` folder. What to insist on should be `"rhel{}-{}".format(redhat_ver, suffix)`.

Further, if you've created a komodoenv on RHEL7, using the same komodoenv on RHEL8 should result in an error, with the possibility of re-creating.

What do you think?

Reply:
:+1: I think checking for `root/` is a good heuristic. I'll add RHEL version checking to this too.

I think it does make sense that `komodoenv` should only be used on the same version of distro on which it was created, so adding a warning to the generated `enable` script would make sense.

Reply:
I've landed on copying `KOMODO_RELEASE`'s suffix. I think it makes sense that `komodoenv` would track `stable-py36-rhel7` rather than just `stable`. We can tack on a update check where if eg. `stable-py27` points to a directory that was last touched X months ago, yield a warning about possible stale environment and suggest doing rebuilding komodoenv by sourcing `/prog/res/komodo/stable` first (maybe it's `py38` at that point).